scripts/common_test.py

```diff
@@ -65,9 +66,92 @@
 )
 
 
+class MockCompiler:
+    def wait(self) -> None: # pylint: disable=missing-docstring
+        pass
+
+
+class MockCompilerContextManager():
+    def __init__(self) -> None:
+        pass
+
+    def __enter__(self) -> MockCompiler:
+        return MockCompiler()
+
+    def __exit__(self, *unused_args: str) -> None:
+        pass
+
+
 class CommonTests(test_utils.GenericTestBase):
     """Test the methods which handle common functionalities."""
 
+    def setUp(self) -> None:
+        super().setUp()
+        self.swap_sys_exit = self.swap(sys, 'exit', lambda _: None)
+        def mock_context_manager() -> MockCompilerContextManager:
+            return MockCompilerContextManager()
+        self.swap_ng_build = self.swap(
+            servers, 'managed_ng_build', mock_context_manager)
+        self.print_arr: list[str] = []
+        def mock_print(msg: str) -> None:
+            self.print_arr.append(msg)
+        self.print_swap = self.swap(builtins, 'print', mock_print)
+
+    def test_run_ng_compilation_successfully(self) -> None:
+        swap_isdir = self.swap_with_checks(
+            os.path, 'isdir', lambda _: True, expected_kwargs=[])
+
+        with self.print_swap, self.swap_ng_build, swap_isdir:
+            common.run_ng_compilation()
+
+        self.assertNotIn(
+            'Failed to complete ng build compilation, exiting...',
+            self.print_arr
+        )
+
+    def test_run_ng_compilation_failed(self) -> None:
+        swap_isdir = self.swap_with_checks(
+            os.path, 'isdir', lambda _: False, expected_kwargs=[])
+
+        with self.print_swap, self.swap_ng_build, swap_isdir:
+            with self.swap_sys_exit:
```

Comment:
Can you assert that `sys.exit` gets called with the right exit code? In general, I think it's good to use `self.swap_with_checks` instead of just `self.swap` wherever possible

Reply:
Done!